plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -615,6 +617,24 @@ public Iterator<TableColumnsMetadata> streamTableColumns(ConnectorSession sessio
                 .iterator();
     }
 
+    private static Schema getCurrentSnapshotSchema(Table icebergTable)
```

Comment:
Let's add javadoc explaining how `getCurrentSnapshotSchema(Table)` is different from `Table#schema()` and when this should be used.